sklearn/metrics/classification.py

```diff
@@ -603,12 +603,19 @@ def f1_score(y_true, y_pred, labels=None, pos_label=1, average='binary',
     The F1 score can be interpreted as a weighted average of the precision and
     recall, where an F1 score reaches its best value at 1 and worst score at 0.
     The relative contribution of precision and recall to the F1 score are
-    equal. The formula for the F1 score is::
+    equal. The formula for the F1 score in the simple ``binary`` case is::
 
         F1 = 2 * (precision * recall) / (precision + recall)
 
-    In the multi-class and multi-label case, this is the weighted average of
-    the F1 score of each class.
+    In the multi-class and multi-label case, the overall F1 score can be
+    calculated as a weighted average of the F1 scores for the classes/labels.
+    Depending on the averaging methods, there are different overall F1 scores::
+    ``macro-averaging`` gives the arithmetic mean of the F1 scores; ``micro-
+    averaging`` uses the total true positives, false negatives and false
```

Comment:
There are many options for calculating a weighted average based on the confusion matrix. With many terms describing the methods, it indeed reads not smooth. I used a minimum description to explain. But mathematically, it is correct. 

Two references: 

1. Van Asch, Vincent. ”Macro-and micro-averaged evaluation measures [[basic
draft]].” (2013). Retrieved from http://www.clips.uantwerpen.be/~vincent/pdf/microaverage.pdf

("3.3 Label-frequency-based micro-averaged" method is the "weighted" option.)

2. Zhang, Min-Ling, and Zhi-Hua Zhou. ”A review on multi-label learning algorithms.” IEEE transactions on knowledge and data engineering 26.8 (2014): 1819-1837.